skimage/morphology/footprints.py

```diff
@@ -95,6 +99,126 @@ def footprint_from_sequence(footprints):
     return morphology.binary_dilation(imag, footprints)
 
 
+class FootprintComponent(tuple):
+    """Part of a decomposed footprint.
+
+    Parameters
+    ----------
+    component : array
+        The footprint component that makes up part of the decomposition.
+    repetitions : int
+        The number of times the `component` should be applied (at least once).
+
+    Notes
+    -----
+    This is a subclass of a tuple and behaves in almost the same way.
+    """
+
+    # Dev notes:
+
+    __slots__ = ()
+
+    def __new__(cls, component, *, repetitions):
+        if not isinstance(component, np.ndarray):
+            msg = (
+                "Expected first item in FootprintComponent to be a numpy.ndarray, "
+                f"got {component!r}"
+            )
+            raise TypeError(msg)
+        if not operator.index(repetitions):
+            msg = (
+                "Expected second item in FootprintComponent to be a positive integer, "
+                f"got {repetitions!r}"
+            )
+            raise TypeError(msg)
+        if repetitions < 1:
+            msg = "`repetitions` must be at least 1"
+            raise ValueError(msg)
+
+        new = super().__new__(cls, (component, repetitions))
+        return new
+
+    @property
+    def component(self):
+        """The footprint component that makes up part of the decomposition."""
+        return self[0]
+
+    def repetitions(self):
+        """The number of times the `component` should be applied."""
+        return self[1]
+
+
+def footprint_rectangle(shape, *, dtype=np.uint8, decomposition="sequence"):
+    """
```

Comment:
Needs summary; perhaps worth including the word `hyperrectangle`.

Reply:
Still needs a summary here.

Reply:
Done.